### DISCUSSION TITLE
NSClient service crash

### DISCUSSION THREAD

**Discussion by manugu87** [mickem/nscp#831] [General]
Hi,
We have many problems with all Windows Servers, nsclient service crash all time.

I upload screenshot of debug nsclient.log, event viewer crash and nsclient.ini, we add 127.0.0.1 and ::1 to allowed host after screenshot.

Other screenshot show correct check on Nagios XI while nsclient service is running.

![Screenshot_4](https://github.com/mickem/nscp/assets/726422/5f4d5cf9-998c-496a-b8b9-5dfd5c9b7f9c)

![Screenshot_2](https://github.com/mickem/nscp/assets/726422/1000aadd-8694-4748-94a6-06ed0ec8c7d8)

![Screenshot_1](https://github.com/mickem/nscp/assets/726422/214c491c-f803-406a-ad14-9735f0c1ba40)

![Screenshot_5](https://github.com/mickem/nscp/assets/726422/d4c0427f-3d36-4ab9-85fb-3852222e947a)

I can't identify what is the problem, it happens to us on most Windows Server computers.

**Comment by mickem** [comment]
Thanks for the screenshots, they help, although they do not yet show the cause.

**What they do show:**

- You are running **0.5.2.x** (the log lines carry the build path `c:\source\0.5.2\`). That is a 2018 build; a lot of the code involved here has been rewritten since, and the current release is 0.21.0. The first thing I would do on a server that crashes is upgrade it, then see whether the crash comes back.
- The log errors are not the crash. All of them come from something **on the server itself** connecting to the NRPE (5666) and check_nt (12489) ports from `127.0.0.1` / `::1`, five times in the same second, and dropping the connection immediately. "Rejected connection" is `allowed hosts` refusing them; "Failed to handle incoming connection: remote_endpoint: El identificador de archivo proporcionado no es válido" is Winsock error 10009 (the peer had already closed the socket when the agent asked for its address). 0.5.2 catches that and carries on, so it is noise. It looks like a local vulnerability scanner or security agent probing open ports; you can leave localhost out of `allowed hosts`, that setting neither causes nor fixes the crash.
- The timing does not match either: the last log line is 13:10:22 and the crash that day is 15:25:39, and the six crashes are days apart. That points at a slow leak or a rare race rather than at one check.
- Event 7034 is only the Service Control Manager reporting that the process died; it never says why.

**What I need to say more:**

1. The **Application** log (not System) around each crash time: Event 1000 "Application Error" names the faulting module and the exception code. Please paste one.
2. The **crash dump**. 0.5.2 archives one per crash in the `crashes` folder under the install directory (`[/settings/crash] archive folder`): a `<guid>.dmp` and a `<guid>.dmp.txt`. The `.txt` is enough to start with.
3. The **last lines of `nsclient.log` before the crash** with `[/settings/log] level = debug`, so we can see which command ran last (Nagios XI sends a burst of checks per host, and the one that is always last before a crash is the suspect).

If you upgrade first, note that current releases write a plain-text `<timestamp>.crash` file instead of the Breakpad dump, naming the exception, the address and the module, and `check_nscp` can alert on it (`check_nscp "crit=crash_age < 7d"`). If it still crashes on a current release, post that file and we have something to fix.
